### PR TITLE
Allow space application supporter to access specific app POST endpoints

### PR DESCRIPTION
**Pull request by philippthun**
Space application supporters can start, stop and restart applications.

Enhance apps request tests to check permissions for all roles.

Partly implements #2280 

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `main` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

**Comment by sweinstein22** [comment]
Hi @philippthun , thanks for the PR! I'm noticing that of the endpoints mentioned in the [associated issue](https://github.com/cloudfoundry/cloud_controller_ng/issues/2280) this one only seems to cover the first three. Namely ` PATCH /v3/apps/:guid/features/revisions` and `PATCH /v3/apps/:guid/relationships/current_droplet` both seem to be missing. Is your intent to submit two different PRs for the issue or is this PR not yet ready for review?

**Comment by philippthun** [comment]
@sweinstein22 As we had to change a lot in the spec file, we decided to have a dedicated PR for the start, stop and restart actions. So we can get some feedback if these changes (i.e. test permissions for all roles) make sense.

**Comment by sweinstein22** [comment]
## Acceptance

Confirmed user only has space_application_supporter and organization_user roles:

```
$ cf curl /v3/roles?user_guids="d8685663-a98a-4c4d-86b0-4523dd45a8b1"
{
  ...
  "resources": [
    {
      "guid": "29ab067c-7205-4fe4-be13-bd63009fcc2a",
      "type": "space_application_supporter",
      "relationships": {
        "user": {
          "data": {
            "guid": "d8685663-a98a-4c4d-86b0-4523dd45a8b1"
          }
        },
      ...
    },
    {
      "guid": "f079f79a-81d9-4ec1-8229-051ddafc3c72",
      "type": "organization_user",
      "relationships": {
        "user": {
          "data": {
            "guid": "d8685663-a98a-4c4d-86b0-4523dd45a8b1"
          }
        },
      ...
    }
  ]
}
```

Ran `cf curl` against `/v3/apps/:guid/actions/stop`, `/v3/apps/:guid/actions/start`, and `/v3/apps/:guid/actions/restart` as user and as a space developer, piped results to files

Confirmed that output of calls was only different in the `updated_at` timestamps:
```
$ diff space_app_supporter_stop space_dev_stop
4c4
<   "updated_at": "2021-06-01T21:48:01Z",
---
>   "updated_at": "2021-06-01T21:40:56Z",
```
```
$ diff space_app_supporter_start space_dev_start
4c4
<   "updated_at": "2021-06-01T21:48:08Z",
---
>   "updated_at": "2021-06-01T21:46:00Z",
```
```
$ diff space_app_supporter_restart space_dev_restart
4c4
<   "updated_at": "2021-06-01T21:48:08Z",
---
>   "updated_at": "2021-06-01T21:46:00Z",
```

✅   Manually ran commands as the `space_application_supporter` and confirmed that app state changed as expected

### Code Review

Most file changes are minimal, picked through the request spec changes in more depth given the significant number of changes.

#### Changes noted:
- Event emission has been moved into its own context
- Addition of permission checks for each endpoint
- Designated `context 'app lifecycle is buildpack'` section in app start section, aligning with other action sections
- In `user_helpers` there's a [correction to an error made previously](https://github.com/cloudfoundry/cloud_controller_ng/pull/2305/files#diff-4eccfd65feea2ca0a3b838852fe7879f645844607b5cf6b1e9547fbede757646L270-R273) that improperly mocked out `untrusted_can_write_to_space` when disallowing users write access 

#### Changes requested:
Super minor/nitpicky, but it feels a little odd to have contexts starting with the phrase "it", given that there are `it` blocks. I Would it feel reasonable to change these to `starting an app`, `stopping an app`, and `restarting an app`?

✅   Docs: Confirmed that documentation for all three endpoints display correctly and have permissions for new role indicated

File: spec/request/apps_spec.rb
Comment (sweinstein22) [inline spec/request/apps_spec.rb:2124]: Super minor/nitpicky, but it feels a little odd to have contexts starting with the phrase "it", given that there are `it` blocks. I Would it feel reasonable to change these to `starting an app`, `stopping an app`, and `restarting an app`?

Comment (sweinstein22) [reply spec/request/apps_spec.rb]: Perfect, thanks for being so responsive. Happy to help unblock you all, thanks for the PR!

**Review by sweinstein22 — APPROVED** [review]

**Pull request merged**
